src/EditorFeatures/CSharpTest/ConflictMarkerResolution/ConflictMarkerResolutionTests.cs

```diff
@@ -529,6 +529,154 @@ static void Main2(string[] args)
             }.RunAsync();
         }
 
+        [WorkItem(23847, "https://github.com/dotnet/roslyn/issues/23847")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsResolveConflictMarker)]
+        public async Task TestTakeTop_TopCommentedOut()
+        {
+            var source = @"
+public class Class1
+{
+    public void M()
+    {
+        /*
+<<<<<<< dest
```

Comment:
💭 I guess we'll need an analyzer to cover the case of markers entirely inside comments.

Reply:
Yup.  But we can wait on that for now IMO.